RFC template (#98501)
elastic/kibana: rfcs/0000_template.md

- Start Date: (fill me in with today's date, YYYY-MM-DD)
- TTL: (e.g. "April 20th, 2021", time the review is expected to be completed by. Don't use relative days.)
- Champion: (usually you, person who writes and updates the draft and incorporates feedback)
- Main reviewer: (somebody familiar with the subject matter, who has committed to provide timely and detailed reviews for this RFC)

> - Owner team: (team who will own this feature, if it is accepted)

Reviewer: nit

```suggestion
- Owner team: (team who will own implementation, if it is accepted)
```

> - Stakeholders: (people or groups who may be affected by the proposed changes)

Reviewer: nit
```suggestion
- Stakeholders: (people or groups who will be affected by the proposed changes)
```

> - RFC PR: (link to PR which implements a PoC of the proposal)

Reviewer: The purpose of this field was changed, and the title "RFC PR" is no longer accurate. Historically, this was a link to the RFC PR itself. For example, https://github.com/elastic/kibana/blob/master/rfcs/text/0012_encryption_key_rotation.md has a "RFC PR" linked to https://github.com/elastic/kibana/pull/72828.

I think we should leave the "RFC PR" field and provide a better description, in addition to adding a new "POC PR" field.

Author: Done.

- Kibana Issue: (link to issue where the proposed feature is tracked)

# Executive Summary

> Summarize this RFC so those unfamiliar with the project and code, and without a
> lot of time, can quickly understand what the problem is, why it is important,
> and the proposed solution. Below are some suggested sections for the Executive
> Summary. Tweak as you desire and try to keep it succinct.

Reviewer: nit

```suggestion
Summarize this RFC so those unfamiliar with the project and code can quickly understand
what the problem is, why it is important,
and the proposed solution. Below are some suggested sections for the Executive
Summary. Tweak as you desire and try to keep it succinct.
```

## Problem statement

What is the problem we are trying to solve? Supply any relevant background
context. Why is this something we should focus on _now_.

Focus on explaining the problem so that if this RFC is not accepted, this
information could be used to develop alternative solutions. In other words,
don't couple this too closely with the solution you have in mind.

## Goals

What are the goals of this project? How will we know if it was successful?

## Proposal

> What are we doing to achieve those goals and solve the problem?

Reviewer: nit
```suggestion
What are we doing to achieve the goals and solve the problem?
```

and include examples of how the feature is used. Any new terminology should be
defined here.

Below are a couple of suggested sections that you can tweak as you desire.

## Terminology

A glossary of new terms can be very helpful.

> ## Architectural diagrams.

Reviewer: nit: +1000 for architectural diagrams. However, this seems to encourage a dedicated section for architectural diagrams, and normally I've seen them included within the detailed design itself.

Reviewer 2: Yea, that makes total sense (I proposed this section initially). I'd be on board with removing it and just maybe adding a sentence under detailed design that says something about encouraging architectural diagrams.

Author: FWIW, it currently also is under "Detailed design", the detailed design has the first-level heading `#` and this section has the second-level heading `##` under "Detailed design".

I moved it as a plain sentence under "Detailed design", as suggested.
